src/components/TextInput/BaseTextInput/types.ts

```diff
@@ -51,15 +51,11 @@ type CustomBaseTextInputProps = {
 
     /**
      * Autogrow input container length based on the entered text.
-     * Note: If you use this prop, the text input has to be controlled
```

Comment:
This does not seem to be true, so I removed this note. controlled text inputs are much less performant so we don't want to do this if we don't have to.